astropy/units/core.py

```diff
@@ -787,7 +801,7 @@ def _normalize_equivalencies(equivalencies):
 
         return normalized
 
-    def __pow__(self, p):
+    def __pow__(self, p: Real) -> CompositeUnit:
```

Comment:
This is a case where the annotation is stricter then what the code actually allows at runtime, but I am not convinced that array powers should be supported at all and if anyone does get spooked into changing a constant-valued array to a scalar in their code because of this annotation then I don't think that's a bad thing at all.